dojo/tools/gitlab_dast/parser.py

```diff
@@ -0,0 +1,163 @@
+    def parse_json(self, file):
+        data = file.read()
+        try:
+            tree = json.loads(str(data, "utf-8"))
+        except:
+            tree = json.loads(data)
+
+        return tree
+
+    def get_items(self, tree, test):
+        items = {}
+
+        # iterating through each vulnerability
+        for node in tree["vulnerabilities"]:
+            item = get_item(node, test)
+
+            item_key = hashlib.sha256(
+                "|".join([item.severity, item.title, item.description]).encode()
+            ).hexdigest()
+
+            if item_key in items:
+                items[item_key].unsaved_endpoints.extend(item.unsaved_endpoints)
+                items[item_key].nb_occurences += 1
+            else:
+                items[item_key] = item
+
+        return list(items.values())
+
+    def get_findings(self, file, test):
+        if file is None:
+            return None
+
+        tree = self.parse_json(file)
+        if tree:
+            return self.get_items(tree, test)
+
+    def convert_severity(self, num_severity):
+        """Convert severity value"""
+        if num_severity >= -10:
+            return "Low"
+        elif -11 >= num_severity > -26:
+            return "Medium"
+        elif num_severity <= -26:
+            return "High"
+        else:
+            return "Info"
+
+
+# iterating through properties of each vulnerability
+def get_item(vuln, test):
+
+    if vuln["category"] != "dast":
+        return None
+
+    # scanner_confidence
+    scanner_confidence = get_confidence_numeric(vuln["confidence"])
+
+    # id
+    if "id" in vuln:
+        unique_id_from_tool = vuln["id"]
+    else:  # deprecated
+        unique_id_from_tool = vuln["cve"]
+
+    # title
+    if "name" in vuln:
+        title = vuln["name"]
+    # fallback to using id as a title
+    else:
+        title = unique_id_from_tool
+
+    # description
+    description = f"Scanner: {vuln['scanner']['name']}\n"
+    if "message" in vuln:
+        description += f"{vuln['message']}\n"
+    elif "description" in vuln:
+        description += f"{vuln['description']}\n"
+
+    # TODO: found_by
+
+    cve = vuln["cve"]
+    cwe = 0
+
+    references = ""
+    for ref in vuln["identifiers"]:
+        if ref["type"].lower() == "cwe":
+            cwe = int(ref["value"])
+        else:
+            references += f"Identifier type: {ref['type']}\n"
+            references += f"Name: {ref['name']}\n"
+            references += f"Value: {ref['value']}\n"
+            if "url" in ref:
+                references += f"URL: {ref['url']}\n"
+            references += "\n"
+
+    finding = Finding(
+        test=test,  # Test
+        nb_occurences=1,  # int
+        unique_id_from_tool=unique_id_from_tool,  # str
+        scanner_confidence=scanner_confidence,  # int
+        title=title,  # str
+        description=description,  # str
+        references=references,  # str (identifiers)
+        cve=cve,  # str
+    )
```

Comment:
We should implement 2 more attributes `static_finding` and `dynamic_finding` (boolean)

Reply:
Done